sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -4365,8 +4365,15 @@ pi_result piEventRelease(pi_event Event) {
     die("piEventRelease: called on a destroyed event");
   }
 
+  // The event is no longer needed upstream, but we have to wait for its completion
+  // in order to do proper cleanup. Otherwise refcount may still be non-zero in the
+  // check below and we will get event leak.
+  if (!Event->CleanedUp)
```

Comment:
I believe the change is conservatively correct.

But I also think that if/when SYCL RT starts doing piEventRelease as soon as possible after it knows event will not be used by itself or the user that their will be a performance drop because
1 – It may affect batching for no good reason.
2- It may cause hostsynchronize when none was intended/wanted.

It is #2 that really concerns me.  So, while I will approve the change in this review, I think we should be looking into
how to fix the leak without doing the piEventsWait here.

Reply:
I have put a TODO comment to address this concern